### PR TITLE
Launcher: Close gracefully when a userDataDir is specified

### LINKED ISSUE #527: using --user-data-dir=profilepath corrupts settings and remove all extensions
Launching with args:
```js
	const browser = await puppeteer.launch({
	  headless: false,
	  args: ['--user-data-dir=/pathtoprofile/'],
	});
```
Will override some files and removes all extensions from the chrome profile + alerts for "settings corrupted" at chrome://settings screen.

### PR DESCRIPTION
**Pull request by JoelEinbinder**
My version of #693 

In order to make tests work, I needed to add a `'close'` event to `Browser`. This broke the doclint, which didn't like browser having a `'close'` event and a `.close` method. So that is why this patch is bigger than it should be.

Fixes #527

### CODE REVIEW

File: lib/Launcher.js [hunk 1]
```diff
@@ -96,18 +101,32 @@ class Launcher {
...
-        process.kill(-chromeProcess.pid);
+      if (temporaryUserDataDir) {
+        // Force kill chrome.
+        if (process.platform === 'win32')
+          childProcess.execSync(`taskkill /pid ${chromeProcess.pid} /T /F`);
+        else
+          process.kill(-chromeProcess.pid, 'SIGKILL');
+        // Attempt to remove temporary profile directory to avoid littering.
+        try {
+          removeSync(temporaryUserDataDir);
```
Comment (JoelEinbinder) [inline lib/Launcher.js:123]: @aslushnikov I copied this from your previous PR. I'm not sure if we need it.

Comment (aslushnikov) [reply lib/Launcher.js:123]: That's to handle "process.exit " case.

File: test/test.js [hunk 2]
```diff
@@ -120,7 +120,9 @@ describe('Browser', function() {
     const options = Object.assign({userDataDir}, defaultBrowserOptions);
     const browser = await puppeteer.launch(options);
     expect(fs.readdirSync(userDataDir).length).toBeGreaterThan(0);
+    const closePromise = new Promise(fulfill => browser.once('close', fulfill));
```
Comment (aslushnikov) [inline test/test.js]: let's make `browser.close()` return a promise

File: lib/Browser.js [hunk 3]
```diff
@@ -60,6 +62,11 @@ class Browser {
   }
 }
 
+/** @enum {string} */
+Browser.Events = {
```
Comment (aslushnikov) [inline lib/Browser.js]: let's kill the event

**Review by aslushnikov — COMMENTED** [review]
Awesome!

**Comment by aslushnikov** [comment]
@JoelEinbinder any updates on this?

**Comment by JoelEinbinder** [comment]
@aslushnikov I removed the event, `browser.close()` returns a promise.

**Review by aslushnikov — APPROVED** [review]
Please, rebaseline so that there are no conflicts

**Comment by JoelEinbinder** [comment]
Rebased

**Pull request merged**